lib/tiny_tds/client.rb

```diff
@@ -37,6 +38,12 @@ def initialize(opts = {})
       if opts[:dataserver].to_s.empty? && opts[:host].to_s.empty?
         raise ArgumentError, 'missing :host option if no :dataserver given'
       end
+
+      @message_handler = opts[:message_handler] || (block_given? ? Proc.new : nil)
```

Comment:
I don't see anywhere where we allow a block to initialize nor is it tested. Can we simplify this to:

```ruby
@message_handler = opts[:message_handler]
```

Reply:
Being able to pass a block to the initializer is feature of this branch. However, there's a bug here that's fixed in the latest push.

Reply:
Can this PR focus on explicit options only first. We can then open up init with a block later.

Reply:
Well... it *could*, but why? It's very useful, and the primary way I've been using this feature. It's much more elegant and idiomatic than passing in an opt, and the cost of this addition is one ternary expression. Doesn't really make sense for that to be separate imo.

Reply:
You caught me... I'd like to not have it at all :) I have an aversion to adding features (any feature) that is superfluous. No where in the tests is this used, so it would be easy to remove later by mistake. But rather than test it to counter that, it is not needed, pure sugar. If we leave it in, on day someone will want the yield to be moved to a different spot for some other reason. I'll have to remind them of this conversation and the circle will go... basically, this is free as in 🐶 and I (the royal we too) would or should only care and feed the things that are truly needed. I just don't see the need here.

Reply:
You missed that I had already added tests for passing a block 😈.

I suppose it's theoretically possible we might want to move it in the future, and if so, what does that matter? Library APIs sometimes change.

However, with this message handling architecture passing a block to the initializer is equivalent to needing to pass in the opt, since the `call`-able is itself passed into C space to be directly invoked. There is not currently any other opportunity to get something into that space safely (that I know of) except the initializer.

Finally, I think that removing it because it's pure sugar is unwise. Fluid ergonomics in libraries is the difference between a library that you enjoy using and a library that you are irritated to use. We all enjoy using libraries that are nice to use, so why wouldn't we also want to be nice to use?